common/Field_Dimensions.hpp

```diff
@@ -56,6 +57,9 @@ struct Field_Dimensions {
     inline Geometry2d::Rect OurHalf() const { return _OurHalf; }
     inline Geometry2d::Rect TheirHalf() const { return _TheirHalf; }
 
+    //We play on a 2D rectangular field, hence the length "4" of the array
+    inline std::array<Geometry2d::Line, 4> FieldBorders() const { return _FieldBorders; }
```

Comment:
For reference, I did a benchmark of by-reference vs by-value for function parameters a while ago.  It's not exactly the same as this case, but pretty similar.

https://gist.github.com/justbuchanan/a1594aead5fda4e6e526